### PR TITLE
[inetstack] SeqNum and AckNum for RST Segments

### PR DESCRIPTION
**Pull request by ppenna**
## Description

This PR partially addresses https://github.com/microsoft/demikernel/issues/235.

### CODE REVIEW

File: src/rust/inetstack/protocols/tcp/peer.rs [hunk 1]
```diff
@@ -596,26 +596,36 @@ impl<const N: usize> Inner<N> {
...
         }
 
-        // grab the queue descriptor based on the incoming.
+        // Retrieve the queue descriptor based on the incoming segment.
         let &qd: &QDesc = match self.addresses.get(&SocketId::Active(local, remote)) {
             Some(qdesc) => qdesc,
             None => match self.addresses.get(&SocketId::Passive(local)) {
                 Some(qdesc) => qdesc,
-                None => return Err(Fail::new(libc::EBADF, "Socket not bound")),
+                None => {
```
Comment (iyzhang) [inline src/rust/inetstack/protocols/tcp/peer.rs:618]: Should we also send a reset here? It seems like this is a similar case to inactive or closed?

**Review by iyzhang — APPROVED** [review]
LGTM, maybe put in some more details about which scenario in the issue that you have covered.

File: src/rust/inetstack/protocols/tcp/peer.rs (hunk 1 above)
Comment (ppenna) [reply src/rust/inetstack/protocols/tcp/peer.rs:618]: I am not sure which case you are talking about.

Comment (ppenna) [reply src/rust/inetstack/protocols/tcp/peer.rs:618]: Not really. This is the case where we have received a segment for a connection that is not established or not in its way to be established. We just ignore the packet and remote will re-transmit if that is a packet that arrived before the SYN.

**Pull request merged**